Gems/ONNX/Code/Source/Clients/Model.h

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/Component/Component.h>
+#include <AzCore/Component/TickBus.h>
+#include <AzCore/Debug/Timer.h>
+#include <AzFramework/StringFunc/StringFunc.h>
+#include <ONNX/ONNXBus.h>
+#include <onnxruntime_cxx_api.h>
+
+#include "upng/upng.h"
+
+#include <algorithm>
+#include <array>
+#include <filesystem>
+#include <string>
+
+namespace ONNX
+{
+    //! Generic ONNX model which can be used to create an inference session and run inferences.
+    class Model
+    {
+    public:
+        Model()
+        {
+        }
+        //! Required params to create session and run inference, passed to Load().
+        struct InitSettings
+        {
+            //! Source of onnx model file.
+            std::wstring m_modelFile = std::wstring{ W_GEM_ASSETS_PATH } + std::wstring{ L"/model.onnx" };
+            std::string m_modelName = ""; //!< Used to create groupings for ImGui dashboard graphs in editor, idea is that the inference runtimes from the same model instance get displayed on the same graph.
+            std::vector<int64_t> m_inputShape; //!< Specifies dimensions of input, eg a vector specifying dimension and magnitude of dimension such as { 1, 1, 28, 28 }.
+            std::vector<int64_t> m_outputShape; //!< Specifies dimensions of output, eg a vector specifying dimension and magnitude of dimension such as { 1, 10 }.
```

Comment:
Can we change this to AZStd::vector or are these fed into the Ort functions and they need an std::vector?

Reply:
Committed